### PR TITLE
Use github workflows for CI

### LINKED ISSUE #434: Q. Multi-targeted unit tests?
Think i've wondered this before after seeing differences between the way that .NET and .NET Core handle validation of characters in paths, but thinking about it again in regards to code page handling with #427:

The unit tests are currently run as .NET Core 2.0 only, but they could be made to multi-target both some version of Core, and .NET 4.5 as well (to match the targets on the lib itself) ?

VS 2019 and the dotnet command line tools will handle that ok (will run under multiple frameworks and display multiple results sets), though the test runner in VS 2017 will only run one of the targets.

### PR DESCRIPTION
**Pull request by piksel**
To enable better feedback and integration, this PR adds Github Actions to build and test merges into master and pull requests.
This includes running on `windows`, `macOS` and `linux` platforms and both on .NET Framework (on windows) and .NET Core.

Fixes #434

_I certify that I own, and have sufficient rights to contribute, all source code and related material intended to be compiled or integrated with the source code for the SharpZipLib open source product (the "Contribution"). My Contribution is licensed under the MIT License._

### CODE REVIEW

**Comment by Numpsy** [comment]
Not directly a question about the CI system, but related and something I've wondered before:

What, at this point, does the manual TestBootstrapper app give you as compared to just using the 

```
dotnet test
```

command on the test project?
The test command should be operable cross platform, and handle multi-targeted tests and such, which seems useful for the test builds?

**Comment by piksel** [comment]
If I recall correctly, it's the report. But whatever reason there used to be, it might not be valid anymore.

**Comment by Numpsy** [comment]
I believe there is an appveyor logger for dotnet test, and looking around I see https://github.com/Tyrrrz/GitHubActionsTestLogger, which sounds like it might be useful for displaying test failures in the Actions output?

**Comment by Numpsy** [comment]
The [actions documentation](https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners) suggests that there is a macOS runner available for builds - might be use to test with that on top on Windows and Linux?

**Comment by Numpsy** [comment]
is the 'Build and Test PR' script running the tests as part of the 'build' task, and then again as part of 'test' task?

File: .github/workflows/on-push.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+          - os: macos-latest
+            libframework: netstandard2
+    steps:
+    - uses: actions/checkout@v2
+      
+    - name: Setup .NET Core
+      if: matrix.libframework == 'netstandard2'
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 2.2.108
```
Comment (Numpsy) [inline .github/workflows/on-push.yml:28]: Is 2.2.108 the appropriate version to use at this point? (I mean, I think it's one of the versions that's out of support, and I think the other PR for the nullability changes would require a v3 SDK for C#8 support)

Comment (piksel) [reply .github/workflows/on-push.yml:28]: Yeah, I think using the latest stable would be more appropriate

**Pull request merged**